alot/commands/globals.py

```diff
@@ -857,8 +857,14 @@ def apply(self, ui):
 
         # Figure out whether we should GPG sign messages by default
         # and look up key if so
-        self.envelope.sign = account.sign_by_default
-        self.envelope.sign_key = account.gpg_key
+        if account.sign_by_default:
+            if account.gpg_key:
+                self.envelope.sign = account.sign_by_default
+                self.envelope.sign_key = account.gpg_key
+            else:
+                msg = 'Cannot find gpg key for account {}'.format(account.address)
+                logging.warn(msg)
+                ui.notify(msg, priority='error')
```

Comment:
That's my understanding as well, prompted by your question about another test I'd written where `detached_signature_for`'s key argument is unset. A nice side effect of this is that with this change we can force key to be set always (since it always will be)